poppy/wfe.py

```diff
@@ -353,6 +353,231 @@ def get_opd(self, wave):
+        If None passed, then the wfe RMS produced is what shows up in PSD calculation.
+        Default to None.
+    incident_angle: astropy quantity
+        Adjusts the WFE based on 
+        Can be passed as either degrees or radians.
+        Default is 0 degrees (paraxial).
+    """
+
+    @utils.quantity_input(wfe=u.nm, radius=u.meter)
+    def __init__(self, name='Model PSD WFE', psd_parameters=None, psd_weight=None, seed=None, 
+                 apply_reflection=False, screen_size=None, rms=None, incident_angle=0*u.deg, 
+                 **kwargs):
+
+        super().__init__(name=name, **kwargs)
+        self.psd_parameters = psd_parameters
+        self.seed = seed
+        self.apply_reflection = apply_reflection
+        self.screen_size = screen_size
+        self.rms = rms
+        
+        # check incident angle units
+        if hasattr(incident_angle, 'unit'):
+            if incident_angle.unit == u.deg:
+                assert incident_angle.value < 90, "Incident angle must be less than 90 degrees."
+            elif incident_angle.unit == u.rad:
+                assert incident_angle.value < (np.pi/2), "Incident angle must be less than pi/2 radians."
+            else:
+                raise Exception('Incident angle units must be either degrees or radians.')
+            self.incident_angle = incident_angle
+        else:
+            raise Exception('Incident angle missing units, must be either degrees or radians.')
+            
+        if psd_weight is None:
+            self.psd_weight = np.ones((len(psd_parameters))) # default to equal weights
+        else:
+            self.psd_weight = psd_weight
+        
+
+    @_check_wavefront_arg
+    def get_opd(self, wave):
+        """
+        Parameters
+        ----------
+        wave : poppy.Wavefront (or float)
+            Incoming Wavefront before this optic to set wavelength and
+            scale, or a float giving the wavelength in meters
+            for a temporary Wavefront used to compute the OPD.
+        """
+        
+        # check that screen size is at least larger than wavefront size
+        wave_size = wave.shape[0]
+        if wave.ispadded is True: # get true wave size if padded to oversample.
+            wave_size = int(wave_size/wave.oversample)
+        
+        # check that screen size exists
+        if self.screen_size is None:
+            self.screen_size = wave.shape[0]
+            
+            if wave.ispadded is False: # sometimes the wave is not padded.
+                self.screen_size = self.screen_size * 4 # default 4x, open for discussion
+        
+        elif self.screen_size < wave_size:
+            raise Exception('PSD screen size smaller than wavefront size, recommend at least 2x larger')
+        
+        # get pixelscale to calculate spatial frequency spacing
+        pixelscale =  * u.pixel # default setting is m/pix, force to meter only
```

Comment:
FYI - This looks to be the offending line that's making the tests fail. Looks like there's a missing variable between the = and * symbols.

Reply:
That line was supposed to get deleted when I changed everything to be wave.pixscale. Let me submit the correction.